src/Jira/Api.php

```diff
@@ -835,4 +846,42 @@ public function closeIssue($issue_key)
 		return $result;
 	}
 
+	 /**
+	 * Returns project components.
+	 *
+	 * @param string $project_key Project key.
+	 *
+	 * @return array|false
+	 * @since  2.0.0
+	 */
+	public function getProjectComponents($project_key)
+	{
+		return $this->api(self::REQUEST_GET, sprintf('/rest/api/2/project/%s/components', $project_key), array(), true);
+	}
+
+	 /**
+	 * Get all issue types with valid status values for a project.
+	 *
+	 * @param string $project_key Project key.
+	 *
+	 * @return array
+	 * @since  2.0.0
+	 */
+	public function getAllIssueTypes($project_key) 
+	{
+		return $this->api(self::REQUEST_GET, "/rest/api/2/project/{$project_key}/statuses", array(), true);
+	}
+
+	 /**
+	 * Returns a list of all resolutions.	
+	 *
+	 * @param string $project_key Project key.
+	 *
+	 * @return array|false
+	 * @since  2.0.0
+	 */
+	public function getResolutions() 
+	{
+		return $this->api(self::REQUEST_GET, "/rest/api/2/resolution", array(), true);
+	}
```

Comment:
There must be an empty line after last function in a class.

Please configure CS check locally to see the issue.